### PR TITLE
Feature/add reducers and actions

### PR DESCRIPTION
**Pull request by AckonSamuel**
## Description:
The Redux Store was configured in the milestone, and actions and reducers for adding and removing books were implemented.

### General requirements:

- [x]  Make sure that there are no linter errors.
- [x]  Make sure that you used correct flow Gitflow
- [x]  Make sure that you documented your work in a professional way

### Milestone requirements

- [x]  Add React Redux (` npm install react-redux`)
- [x]  Structure your application's files using a "feature folder" approach and use the ducks pattern for your Redux files.
- [x]  Write your book's actions and reducer (in `/src/redux/books/books.js`) using the ducks pattern.
- [x] Write your categories actions and reducer (in `/src/redux/categories/categories.js`) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- [x] Configure the Redux Store (/src/redux/configureStore.js)

<hr>
 Happy Reviewing 🐣

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { createStore, combineReducers } from 'redux';
```
Comment (codecaiine) [inline src/redux/configureStore.js]: - [x] Good job so far but I would like to recommend you to import `configureStore `from `@reduxjs/toolkit` as required .  `configureStore ` is a friendly abstraction over the standard Redux `createStore` function that adds good defaults to the store setup for a better development experience. Please fix it :+1:

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        ...state, action.payload,
+      ];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.payload.id);
+    default: return state;
+  }
+};
+
+export { AddBook, RemoveBook };
+export default BookReducer;
```
Comment (codecaiine) [inline src/redux/books/books.js]: - [x] Good job so far to implement some of the requirements but some elements are missing to make your implementation complete :cry: Don't forget that you need to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux). Please fix it :smile: . See the perfect example below :arrow_down: 
```
// widgets.js

// Actions
const LOAD   = 'my-app/widgets/LOAD';
const CREATE = 'my-app/widgets/CREATE';
const UPDATE = 'my-app/widgets/UPDATE';
const REMOVE = 'my-app/widgets/REMOVE';

// Reducer
export default function reducer(state = {}, action = {}) {
  switch (action.type) {
    // do reducer stuff
    default: return state;
  }
}

// Action Creators
export function loadWidgets() {
  return { type: LOAD };
}

export function createWidget(widget) {
  return { type: CREATE, widget };
}

export function updateWidget(widget) {
  return { type: UPDATE, widget };
}

export function removeWidget(widget) {
  return { type: REMOVE, widget };
}

// side effects, only as applicable
// e.g. thunks, epics, etc
export function getWidget () {
  return dispatch => get('/widget').then(widget => dispatch(updateWidget(widget)))
}
```

File: src/reportWebVitals.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
-const reportWebVitals = onPerfEntry => {
+const reportWebVitals = (onPerfEntry) => {
   if (onPerfEntry && onPerfEntry instanceof Function) {
-    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+    import('web-vitals').then(({
+      getCLS, getFID, getFCP, getLCP, getTTFB,
+    }) => {
       getCLS(onPerfEntry);
       getFID(onPerfEntry);
       getFCP(onPerfEntry);
```
Comment (codecaiine) [inline src/reportWebVitals.js]: - [ ] [OPTIONAL]  Please remove all unnecessary files and folder in your project in order to make it clean :+1:

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @AckonSamuel , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good project implementation :heavy_check_mark: 
- Descriptive PR :heavy_check_mark: 
- No linters errors :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (AckonSamuel) [reply src/redux/configureStore.js]: Done!

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @AckonSamuel, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**